doc/examples/fast_streamline_search.py

```diff
@@ -0,0 +1,202 @@
+The same list of streamlines can be given to recover the self distance matrix.
+
+    FastStreamlinesSearch Class
+    - ref_streamlines : reference streamlines, that will be searched in (tree)
+    - max_radius : is the maximum distance that can be used with radius search
+
+    radius_search() function
+    - radius : for each streamline search find all similar ones in the
+        "ref_streamlines" that are within the given radius
+
+    Be cautious, a large radius might result in a dense distance computation,
+    requiring a large amount of time and memory.
+    Recommended range of the radius is from 1 to 10 mm.
+"""
+
+l21_radius = 7.0
+fss = FastStreamlineSearch(ref_streamlines=model_af_l, max_radius=l21_radius)
+coo_mdist_mtx = fss.radius_search(streamlines, radius=l21_radius)
+
+"""
+Extract indices of streamlines with an similar ones in the reference
+"""
+ids_s = np.unique(coo_mdist_mtx.row)
+ids_ref = np.unique(coo_mdist_mtx.col)
+
+recognized_af_l = streamlines[ids_s]
+
+"""
+let's visualize streamlines similar to the Arcuate Fasciculus Left bundle
+"""
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+scene.add(actor.line(model_af_l, colors=(0, 1, 0)))
+scene.add(actor.line(recognized_af_l, colors=(0, 0, 1)))
+scene.set_camera(focal_point=(-18.17281532, -19.55606842, 6.92485857),
+                 position=(-360.11, -30.46, -40.44),
+                 view_up=(-0.03, 0.028, 0.89))
+if interactive:
+    window.show(scene)
+else:
+    window.record(scene, out_path='AF_L_recognized_bundle.png',
+                  size=(600, 600))
+
+"""
+.. figure:: AF_L_recognized_bundle.png
+   :align: center
+
+   Recognized Arcuate Fasciculus Left bundle
+
+"""
+
+"""
+Color the resulting AF by the nearest streamlines distance
+"""
+
+nn_s, nn_ref, nn_dist = nearest_from_matrix_row(coo_mdist_mtx)
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+cmap = actor.colormap_lookup_table(scale_range=(nn_dist.min(), nn_dist.max()))
+scene.add(actor.line(recognized_af_l, colors=nn_dist, lookup_colormap=cmap))
+scene.add(actor.scalar_bar(cmap, title="distance to atlas (mm)"))
+scene.set_camera(focal_point=(-18.17281532, -19.55606842, 6.92485857),
+                 position=(-360.11, -30.46, -40.44),
+                 view_up=(-0.03, 0.028, 0.89))
+if interactive:
+    window.show(scene)
+else:
+    window.record(scene, out_path='AF_L_recognized_bundle_dist.png',
+                  size=(600, 600))
+
+"""
+.. figure:: AF_L_recognized_bundle_dist.png
+   :align: center
+
+   Recognized Arcuate Fasciculus Left bundle colored by distance to ref
+
+"""
+
+"""
+Display the streamlines reference reached
+"""
+
+ref_in_reach = np.zeros([len(model_af_l)], dtype=bool)
+ref_in_reach[ids_ref] = True
+
+scene = window.Scene()
+scene.SetBackground(1, 1, 1)
+if np.any(ref_in_reach):
+    scene.add(actor.line(model_af_l[ref_in_reach], colors=(0, 1, 0)))
+if np.any(~ref_in_reach):
+    scene.add(actor.line(model_af_l[~ref_in_reach], colors=(1, 0, 0)))
```

Comment:
this might be too advanced for a tutorial. Many users do not understand `any` and even less `~`.  Too many negation, can you find an alternative?

Reply:
its simply to separate in 2 groups, however it needed a check to avoid empty array.
I changed it to only change the color, it should be simplier now.